src/entities/FactorOperationalMemory.jl

```diff
@@ -109,52 +77,52 @@ DevNotes
-  # which index is being solved for in params?
-  varidx::Int 
-  # FIXME make type stable, JT should now be type stable if rest works
-  #   user defined measurement values for each approxConv operation
-  measurement::Vector{MT} 
-  # TODO refactor and deprecate this old approach, Union{Type{SingleThreaded}, Type{MultiThreaded}}
-  threadmodel::Type{<:_AbstractThreadModel} 
-  # FIXME, deprecate for only `readonly` and build CalcFactor objects on stack instead
-    ## will be obsolete: particular convolution computation values per particle idx (varies by thread)
-  cpt::Vector{<:ConvPerThread}
-  # inflationSpread
-  inflation::Float64
-  # Which dimensions does this factor influence.  Sensitive (mutable) to both which 'solvefor index' variable and whether the factor is partial dimension
-  partialDims::Vector{<:Integer}
-  # variable types for points in params
-  vartypes::Vector{DataType}
-  # experimental feature to embed gradient calcs with ccw
-  _gradients::G
-  # type used for cache
-  dummyCache::CT
-  #
-  fullvariables::Vector{DFGVariable}
+mutable struct CommonConvWrapper{T<:AbstractFactor,NTP<:Tuple,G,MT,CT} <: FactorOperationalMemory
+    #
+    ### Values consistent across all threads during approx convolution
+    usrfnc!::T # user factor / function
+    # general setup
+    xDim::Int
+    zDim::Int
+    # is this a partial constraint as defined by the existance of factor field `.partial::Tuple`
+    partial::Bool
+    # multi hypothesis settings #NOTE no need for a parameter as type is known from `parseusermultihypo`
+    hypotheses::Union{Nothing,Distributions.Categorical{Float64,Vector{Float64}}}
+    # categorical to select which hypothesis is being considered during convolution operation
+    certainhypo::Union{Nothing,Vector{Int}}
+    nullhypo::Float64
+    # parameters passed to each hypothesis evaluation event on user function, #1321
+    params::NTP # TODO rename to varValsLink::NTP
+    # which index is being solved for in params?
+    varidx::Int
+    # FIXME make type stable, JT should now be type stable if rest works
+    #   user defined measurement values for each approxConv operation
+    measurement::Vector{MT}
+    # TODO refactor and deprecate this old approach, Union{Type{SingleThreaded}, Type{MultiThreaded}}
+    threadmodel::Type{<:_AbstractThreadModel}
+    # inflationSpread
+    inflation::Float64
+    # Which dimensions does this factor influence.  Sensitive (mutable) to both which 'solvefor index' variable and whether the factor is partial dimension
+    partialDims::Vector{<:Integer}
+    # variable types for points in params
+    vartypes::Vector{DataType}
+    # experimental feature to embed gradient calcs with ccw
+    _gradients::G
+    # type used for cache
+    dummyCache::CT
+
+    #Consolidation from FMD
+    fullvariables::Vector{DFGVariable}
+
+    #Consolidation from CPT
+    # the actual particle being solved at this moment
+    particleidx::Int
```

Comment:
these Old CPT fields are specific to each particle (thread) in the future so can therefore not stay in CCW.  They should only be created in CalcFactor.

Reply:
We can discuss it at some stage. The mission here was to remove CPT.

Reply:
sounds good